apps/src/music/views/SoundsPanel2.tsx

```diff
@@ -0,0 +1,487 @@
+  folder,
+  sound,
+  onSelect,
+  onPreview,
+  currentSoundRefCallback,
+}) => {
+  const soundPath = folder.id + '/' + sound.src;
+  const isSelected = soundPath === currentValue;
+  const isPlayingPreview = playingPreview === soundPath;
+
+  const onSoundSelect = useCallback(() => {
+    if (!isPlayingPreview) {
+      onPreview(soundPath);
+    }
+    onSelect(soundPath);
+  }, [isPlayingPreview, onPreview, onSelect, soundPath]);
+
+  const onSoundClick = useCallback(
+    (e: React.MouseEvent<HTMLDivElement>) => {
+      e.stopPropagation();
+      onSoundSelect();
+    },
+    [onSoundSelect]
+  );
+
+  return (
+    <div
+      className={classNames(
+        'sounds-panel-sound-entry',
+        styles.soundEntry,
+        isSelected && styles.soundEntrySelected
+      )}
+      onClick={onSoundClick}
+      onKeyDown={event => {
+        if (event.key === 'Enter') {
+          onSoundSelect();
+        }
+      }}
+      ref={isSelected ? currentSoundRefCallback : null}
+      aria-label={sound.name}
+      tabIndex={0}
+      role="button"
+    >
+      <div className={styles.soundEntryMain}>
+        <div
+          className={classNames(
+            styles.iconContainer,
+            SoundStyle[sound.type]?.classNameBackground,
+            SoundStyle[sound.type]?.classNameBorder,
+            isSelected && styles.iconContainerSelected
+          )}
+        >
+          <FontAwesomeV6Icon
+            iconName={SoundStyle[sound.type]?.icon || ''}
+            className={classNames(
+              styles.typeIcon,
+              SoundStyle[sound.type]?.classNameColor
+            )}
+          />
+        </div>
+        <div
+          className={classNames(
+            styles.name,
+            sound.type === 'vocal' && styles.nameVocal
+          )}
+        >
+          {sound.name}
+        </div>
+      </div>
+    </div>
+  );
+};
+
+// ScrollIntoView: This component does an eased scroll of the container's content,
+// starting from a distance below the top, and scrolling to the top.  It's useful
+// to show the user that some new content is scrollable.
+
+interface ScrollIntoViewProps {
+  id?: string;
+  className?: string;
+  doScroll: boolean;
+  delay: number;
+  frames: number;
+  distanceY: number;
+  children: React.ReactNode;
+}
+
+const ScrollIntoView: React.FunctionComponent<ScrollIntoViewProps> = ({
+  id,
+  className,
+  doScroll,
+  delay,
+  frames,
+  distanceY,
+  children,
+}) => {
+  const scrollStep = useRef<number | undefined>(0);
+  const lastScrollPosition = useRef<number | undefined>(undefined);
+  const containerRef: React.MutableRefObject<HTMLDivElement | null> =
+    useRef(null);
+
+  const containerRefCallback = (ref: HTMLDivElement) => {
+    containerRef.current = ref;
+  };
+
+  function easeOutSine(x: number): number {
+    return Math.sin((x * Math.PI) / 2);
+  }
+
+  // Initial render.
+  useEffect(() => {
+    if (!doScroll) {
+      return;
+    }
+
+    setInterval(() => {
```

Comment:
This is fine and _probably_ doesn't need to be changed, but just noting that animating things in an interval is a little problematic - intervals aren't guaranteed to fire at the requested time, they're only guaranteed not to fire _before_ the requested time. So if the main thread is locked up doing other things, this could result in a very slow animation.

If this were rewritten to use [requestAnimationFrame](https://developer.mozilla.org/en-US/docs/Web/API/window/requestAnimationFrame) we'd be able to guarantee the exact duration of the animation, where it'd drop frames if necessary to fit the timeframe.

Personally, I think [react-interpolation-animation](https://www.npmjs.com/package/react-interpolation-animation) works very nicely for this, but of course I'm biased. 😅

Reply:
Interesting. I've used `requestAnimationFrame` bunch for big canvas animations but good point about using it for this kind of animation as well.  Let's keep an eye on animation quality and maybe do this as a follow-up.